pageserver/compaction/src/helpers.rs

```diff
@@ -120,6 +138,12 @@ impl<'a, E: CompactionJobExecutor> LazyLoadLayer<'a, E> {
             Self::Unloaded(dl) => dl.key_range().start,
         }
     }
+    fn lsn(&self) -> Lsn {
```

Comment:
better named current_key_lsn, or return (Key, Lsn) for the `key` function, instead of `lsn`. `lsn` can mean the largest LSN in the delta layer, which might be different from the current key LSN.

Reply:
it's the same situation as for the key function: the delta layer always has a key range but only a single key is returned. I will rename both functions for clarity.